test/dynamo/test_misc.py

```diff
@@ -13286,6 +13287,208 @@ class C:
         f(x)
         self.assertEqual(counter.frame_count, 2)
 
+    def test_datetime_now_in_graph_no_recompile(self):
+        counter = CompileCounter()
+
+        @torch.compile(backend=counter, fullgraph=True)
+        def fn(x):
+            t = datetime.datetime.now()
+            return x + t.second
+
+        x = torch.randn(4)
+
+        # multiple runs shouldn't trigger recompile
+        out1 = fn(x)
+        out2 = fn(x)
```

Comment:
nit: you can use `torch.compiler.set_stance("fail_on_recompile")` on the second call to error on recompiles